src/util/json.cpp

```diff
@@ -162,3 +164,48 @@ void jsont::swap(jsont &other)
   other.object.swap(object);
   other.value.swap(value);
 }
+
+bool operator==(const jsont &left, const jsont &right)
+{
+  if(left.kind != right.kind)
+    return false;
+  switch(left.kind)
+  {
+  case jsont::kindt::J_NULL:
+    return true;
+  case jsont::kindt::J_TRUE:
+    return true;
+  case jsont::kindt::J_FALSE:
+    return true;
+  case jsont::kindt::J_NUMBER:
+    return left.value == right.value;
+  case jsont::kindt::J_STRING:
+    return left.value == right.value;
+  case jsont::kindt::J_ARRAY:
+  {
+    const auto &left_array = static_cast<const json_arrayt &>(left);
+    const auto &right_array = static_cast<const json_arrayt &>(right);
+    return left_array.size() == right_array.size() &&
+           std::equal(
+             left_array.begin(), left_array.end(), right_array.begin());
+  }
+  case jsont::kindt::J_OBJECT:
+  {
+    const auto &left_object = static_cast<const json_objectt &>(left);
+    const auto &right_object = static_cast<const json_objectt &>(right);
+    const size_t left_size =
+      narrow<size_t>(std::distance(left_object.begin(), left_object.end()));
+    const size_t right_size =
+      narrow<size_t>(std::distance(right_object.begin(), right_object.end()));
+    if(left_size != right_size)
+      return false;
+    return std::all_of(
+      left_object.begin(),
+      left_object.end(),
+      [&](const std::pair<std::string, jsont> &pair) {
+        return right_object[pair.first] == pair.second;
```

Comment:
That's an unnecessary lookup when instead you could just use two iterators over the two containers.

Reply:
Yeah, but if someone decides to swap `std::map` for `std::unordered_map`, it could be difficult to debug, so this is more stable/doesn't rely on any invariants.

Reply:
I do wonder which containers would exhibit differences in strict equality testing? But I suppose it is conceivable that two `std::unordered_map` had their elements constructed in a different order, yet should compare equal. So, yes, keep it as is.

Reply:
Personally I still favour 
```
// This relies on ordered maps:
static_assert(std::is_same<typeof(jsont::object), std::map<std::string, jsont>>)
```